icaruscode/CRT/CRTTPCTruthEffMCKalman_module.cc

```diff
@@ -0,0 +1,1721 @@
+				all_crt_candidate_z.push_back(thiscand.crt_hit_pos.Z());
+
+				all_crt_regions.push_back((int)fCrtutils->AuxDetRegionNameToNum(thiscand.thishit.tagger));
+				crt_timestamp.push_back(thiscand.crtTime);
+
+				crt_start_dca.push_back(thiscand.simpleDCA_startDir);	
+				crt_end_dca.push_back(thiscand.simpleDCA_endDir);	
+
+				all_crt_hitcode_map.push_back(i);//Gives ID to each hit for the FEB+channel+pes breakdown below to allow cross-referencing between vectors
+
+				all_crt_hitflags.push_back(thiscand.thishit.ts0_s_corr);
+
+			    	std::map< uint8_t, std::vector<std::pair<int,float> > > thishit_pes = crtcandidate.pesmap; ///< Saves signal hit information (FEB, local-channel and PE) .
+				for(uint8_t k=1; k<232; k++){
+					if(thishit_pes[k].size()>0){
+						for(int j=0; j<(int)thishit_pes[k].size(); j++){
+							all_crt_mac5s.push_back(k);
+							all_crt_hitcode.push_back(i);
+							all_crt_chans.push_back(thishit_pes[k][j].first);
+							all_crt_pes.push_back(thishit_pes[k][j].second);
+						}//end for(int j=0; j<crt_pes[k].size(); j++)
+					}//end if(thishit_pes[k].size()>0)
+				}//end loop searching for FEBs connected to the hit
+			}//end loop over CRT candidates
+		}//end if(num_crt_candidates>0)
+
+	    }//end if(closest.dca != -99999)
+
+	  } // DCA check
+
+		if(track_trueID==-1) num_tracks_badID++;
+		if(track_length>20&&track_trueID==-1) { 
+
+			std::cout << "Event #" << fEvent << " has a track with TrueID=-1 and a length over 20 cm\n";
+			std::cout << "Track length: " << track_length << "\n";
+			num_tracks_badID_greaterthan_20cm++;
+		}//end if(track_length>20&&track_trueID==-1)
+
+   	    /*if(trackList[track_i]->Length()>20)*/ tr_crttpc->Fill();
+	} // Loop over tracks  
+	
+      } // validity check
+
+    } // all track labels in the product
+	std::cout << "Found " << num_tracks_badID << " tracks with bad ID, " << num_tracks_badID_greaterthan_20cm << " of which are >20cm in reconstructed track length\n";
+	std::cout << "End event #" << fEvent << "\n000000000000000000000000000000000000000000000000000000000000000000\n";
+  } // CRTTPCTruthEffMCKalman::produce()
+
+
+void icarus::CRTTPCTruthEffMCKalman::GetAncestorID(int trueid, int &motherid, int &ancestorid, int &layers, std::map<int, simb::MCParticle> all_particles){
+
+	int size_part = (int)all_particles.size();
+	if(trueid<size_part) ancestorid = all_particles[trueid].Mother();
+	else ancestorid = trueid;
+	motherid = ancestorid;
+	int last_ancestorid;
+	bool hitbottom = false;
+	int counter = 0;
+	while(ancestorid%10000000!=0 && counter<1000 && !hitbottom){
+		last_ancestorid = ancestorid;
+		if(ancestorid<size_part) ancestorid = all_particles[ancestorid].Mother();
+		hitbottom = last_ancestorid == ancestorid;
+		counter++;
+	}//end while loop 
+	layers = counter;
+
+
+}//CRTTPCTruthEffMCKalman::GetAncestorID
+void icarus::CRTTPCTruthEffMCKalman::getTickRange(std::vector<art::Ptr<recob::Hit>> hits, int &mintick, int &maxtick){
```

Comment:
Similar comment as above. If a function has "Get" in its name, I expect a return value. Without a deeper dive, I'm not sure whether it makes more sense to add a return value here or to just rename the function. I'll leave that up to you.